rcl/src/rcl/node.c

```diff
@@ -387,6 +404,16 @@ rcl_node_fini(rcl_node_t * node)
       result = RCL_RET_ERROR;
     }
   }
+  rcl_ret = rcl_node_type_description_service_fini(node);
+  if (rcl_ret != RCL_RET_OK && rcl_ret != RCL_RET_NOT_INIT) {
+    RCL_SET_ERROR_MSG("Unable to fini ~/get_type_description service for node.");
+    result = RCL_RET_ERROR;
+  }
+  rcl_ret = rcl_node_type_cache_fini(node);
+  if (rcl_ret != RCL_RET_OK && rcl_ret != RCL_RET_NOT_INIT) {
+    RCL_SET_ERROR_MSG("Unable to fini type cache for node.");
+    result = RCL_RET_ERROR;
+  }
```

Comment:
And the same here; we should only teardown if we initialized to begin with.

Reply:
Sorry - you're right, it will still leave an error state that will add spam in _other_ failure teardown. Will address